plugins/janus_audiobridge.c

```diff
@@ -1503,17 +1535,35 @@ void janus_audiobridge_incoming_rtp(janus_plugin_session *handle, int video, cha
 		}
 		/* Enqueue the decoded frame */
 		janus_mutex_lock(&participant->qmutex);
-		g_queue_push_tail(participant->inbuf, pkt);
-		/* Keep the queue at max of 150 packets, though */
-		if(g_queue_get_length(participant->inbuf) > 150) {
-			/* Remove the oldest packet */
-			janus_audiobridge_rtp_relay_packet *pkt = g_queue_pop_head(participant->inbuf);
-			if(pkt != NULL) {
-				if(pkt->data)
-					g_free(pkt->data);
-				pkt->data = NULL;
-				g_free(pkt);
-				pkt = NULL;
+		/* Insert packets sorting by sequence number */
+		participant->inbuf = g_list_insert_sorted(participant->inbuf, pkt, &janus_audiobridge_rtp_sort);
+		if(participant->prebuffering) {
+			/* Still pre-buffering: do we have enough packets now? */
+			if(g_list_length(participant->inbuf) == DEFAULT_PREBUFFERING) {
+				participant->prebuffering = FALSE;
+				JANUS_LOG(LOG_VERB, "Prebuffering done! Finally adding the user to the mix\n");
+			} else {
+				JANUS_LOG(LOG_VERB, "Still prebuffering (got %d packets), not adding the user to the mix yet\n", g_list_length(participant->inbuf));
+			}
+		} else {
+			/* Make sure we're not queueing too many packets: if so, get rid of the older ones */
+			if(g_list_length(participant->inbuf) >= DEFAULT_PREBUFFERING*2) {
```

Comment:
Yes, the drops are heard by the receiving end. To be clear we were definitely getting complains about drops in the audio before upping to 48khz however I think this might be more frequent now at the higher quality, which makes some sense.

Up until last week we were using the default quality which I believe is 16khz and this still exhibits the issue - you probably hear this on meetecho occasionally? Perhaps it could drop the quality in increments in an ideal world...